src/App.tsx

```
@@ -144,6 +154,130 @@ function App() {
+        // Remove or replace path separators and traversal attempts
+        let sanitized = filename
+          .replace(/[\/\\]/g, '_')           // Replace path separators with underscore
+          .replace(/\.\./g, '_')             // Replace .. with underscore
+          .trim();                           // Remove leading/trailing whitespace
+
+        // Remove leading dots
+        sanitized = sanitized.replace(/^\.+/, '');
+
+        // Remove invalid filename characters (OS-specific but being conservative)
+        // Keep: alphanumeric, spaces, hyphens, underscores, parentheses, periods
+        sanitized = sanitized.replace(/[<>:"|?*\x00-\x1F]/g, '_');
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)